src/Microsoft.OData.Client/ALinq/ResourceBinder.cs

```diff
@@ -289,7 +294,7 @@ private static Expression AnalyzePredicate(MethodCallExpression mce, ClientEdmMo
                     input.ConvertKeyToFilterExpression();
                 }
 
-                if (keyPredicates == null)
+                if (keyPredicates == null || idPredicateGeneratesFilterQueryOption)

```

Comment:
Why do you need to check `idPredicateGeneratesFilterQueryOption` in this case?

Reply:
In both conditions, we need to convert Key to filter expression, incase there was one.